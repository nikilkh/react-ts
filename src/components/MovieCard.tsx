import { useNavigate } from "react-router-dom";
import { movieData } from "./context";

type props = {
  movie: movieData
}
export default function MovieCard(props: props) {
    let navigate = useNavigate();

    function handleBookNow() {
navigate("/bookingpage", {state: props})
    }

  return (
    <>
      <h3>{props.movie.title}</h3>
      <button onClick={handleBookNow}>Book Now</button>
    </>
  );
}
