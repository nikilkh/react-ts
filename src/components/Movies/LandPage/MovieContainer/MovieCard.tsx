import { useNavigate } from "react-router-dom";
import { movieData } from "../../../context";
import { BookNowButtonStyle } from "../LandPage.styled";

type props = {
  movie: movieData;
};
export default function MovieCard(props: props) {
  let navigate = useNavigate();

  function handleBookNow() {
    navigate(`/book/${props.movie.id}/${props.movie.title}`, { state: props });
  }

  return (
    <>
      <h3>{props.movie.title}</h3>
      <BookNowButtonStyle onClick={handleBookNow}>Book Now</BookNowButtonStyle>
    </>
  );
}
