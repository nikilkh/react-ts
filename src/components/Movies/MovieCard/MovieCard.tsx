import React from "react";
import { useNavigate } from "react-router-dom";
import { Button } from "../../Button";
import { props } from "./";

export const MovieCard = (props: props) => {
  const navigate = useNavigate();

  function handleBookNow() {
    navigate(`/book/${props.movie.id}/${props.movie.title}`, { state: props });
  }

  return (
    <>
      <h3>{props.movie.title}</h3>
      <Button onclick={handleBookNow}>Book Now</Button>
    </>
  );
};
