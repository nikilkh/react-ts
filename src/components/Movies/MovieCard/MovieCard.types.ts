import { movieData } from "../../Context/context";

export type props = {
    movie: movieData;
  };