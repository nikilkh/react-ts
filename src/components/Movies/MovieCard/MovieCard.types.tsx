import { movieData } from "../../Context/context";

export type movieCardProps = {
    movie: movieData;
  };