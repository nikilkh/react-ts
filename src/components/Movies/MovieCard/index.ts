export * from "./MovieCard";
export * from "./MovieCard.types"