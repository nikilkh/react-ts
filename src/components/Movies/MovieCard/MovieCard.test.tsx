import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { MovieCard } from "./MovieCard"
const movie = {
  id: 0,
  title: "",
  poster_path: "",
};
const handleBookNow = jest.fn();

describe("<MovieCard/>", () => {
  it("should take snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <MovieCard movie={movie} />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test("book now button", ()=> {
    const button = screen.getby
  })
});
