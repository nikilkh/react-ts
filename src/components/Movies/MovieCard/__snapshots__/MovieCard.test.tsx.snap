// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard/> should take snapshot 1`] = `
<div>
  <h3 />
  <button
    class="sc-bczRLJ cgVelk"
  >
    Book Now
  </button>
</div>
`;
